test/e2e/helpers.go

```diff
@@ -823,3 +829,54 @@ func getPodLogs(ctx context.Context, clientset *kubernetes.Clientset, pod corev1
 	}
 	return b.String()
 }
+
+func InstallHelmChart(ctx context.Context, input clusterctl.ApplyClusterTemplateAndWaitInput, repoURL, chartName, releaseName string, values []string) {
+	clusterProxy := input.ClusterProxy.GetWorkloadCluster(ctx, input.ConfigCluster.Namespace, input.ConfigCluster.ClusterName)
+	kubeConfigPath := clusterProxy.GetKubeconfigPath()
+	settings := helmCli.New()
+	settings.KubeConfig = kubeConfigPath
+	actionConfig := new(helmAction.Configuration)
+	err := actionConfig.Init(settings.RESTClientGetter(), "default", "secret", Logf)
+	Expect(err).To(BeNil())
+	i := helmAction.NewInstall(actionConfig)
+	i.RepoURL = repoURL
+	i.ReleaseName = releaseName
+	Eventually(func() error {
+		cp, err := i.ChartPathOptions.LocateChart(chartName, helmCli.New())
+		if err != nil {
+			return err
+		}
+		p := helmGetter.All(settings)
+		valueOpts := &helmVals.Options{}
+		valueOpts.Values = values
+		vals, err := valueOpts.MergeValues(p)
+		if err != nil {
+			return err
+		}
+		chartRequested, err := helmLoader.Load(cp)
+		if err != nil {
+			return err
+		}
+		release, err := i.RunWithContext(ctx, chartRequested, vals)
+		if err != nil {
+			return err
+		}
+		Logf(release.Info.Description)
+		return nil
+	}, input.WaitForControlPlaneIntervals...).Should(Succeed())
+}
+
+func podListHasNumPods(numPods int) func(pl *corev1.PodList) error {
```

Comment:
is the reason we need to do this because the helper func doesn't actually check the number of pods?

Reply:
This reformulated approach is based on trying to re-use existing capi E2E foo.

`framework.WaitForPodListCondition` is the most appropriate existing thing that I could find, and it takes a `framework.PodListCondition` to be able to enforce what we're looking for. In this particular test case validating that the resultant pod list has n number of running pods is what we want.